src/Symfony/Bundle/TwigBundle/DependencyInjection/Compiler/ExtensionPass.php

```diff
@@ -42,5 +42,16 @@ public function process(ContainerBuilder $container)
         if ($container->has('fragment.handler')) {
             $container->getDefinition('twig.extension.httpkernel')->addTag('twig.extension');
         }
+
+        if ($container->has('assets.packages')) {
+            $container->getDefinition('twig.extension.assets')->addTag('twig.extension');
+        } elseif ($container->has('templating.asset.packages')) {
```

Comment:
The alternative solution is to have the new extension providing a BC layer for the ``asset()`` extension